tests/lib/rules/define-macros-order.js

```diff
@@ -0,0 +1,392 @@
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup>
+          defineProps({
+            test: Boolean
+          })
+          defineEmits(['update:test'])
+          console.log('test')
+        </script>
+      `,
+      options: optionsPropsFirst
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup lang="ts">
+          const props = withDefaults(defineProps<Props>(), {
+            msg: 'hello',
+            labels: () => ['one', 'two']
+          })
+          const emit = defineEmits<{(e: 'update:test'): void}>()
+          console.log('test')
+        </script>
+      `,
+      parserOptions: {
+        parser: require.resolve('@typescript-eslint/parser')
+      }
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup lang="ts">
+          defineEmits(['update:test'])
+          console.log('test')
+        </script>
+      `,
+      options: optionsPropsFirst
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup lang="ts">
+          defineProps({
+            test: Boolean
+          })
+          console.log('test')
+        </script>
+      `,
+      options: optionsPropsFirst
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup lang="ts">
+          console.log('test')
+        </script>
+      `,
+      options: optionsPropsFirst
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup lang="ts">
+          import { bar } from 'foo'
+          defineEmits(['update:test'])
+          defineProps({
+            test: Boolean
+          })
+          console.log('test')
+        </script>
+      `,
+      options: optionsEmitsFirst
+    }
+  ],
+  invalid: [
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup>
+          import { bar } from 'foo'
+          console.log('test1')
+          defineEmits(['update:test'])
+        </script>
+      `,
+      output: `
+        <script setup>
+          import { bar } from 'foo'
+          defineEmits(['update:test'])
+          console.log('test1')
+        </script>
+      `,
+      options: optionsEmitsFirst,
+      errors: [
+        {
+          message: message('defineEmits'),
+          line: 5
+        }
+      ]
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        <script setup>
+          import { bar } from 'foo'
+
+          console.log('test1')
+          defineEmits(['update:test'])
+          console.log('test2')
+          defineProps({
+            test: Boolean
+          })
+          console.log('test3')
+        </script>
+      `,
+      output: `
+        <script setup>
+          import { bar } from 'foo'
+
+          defineProps({
+            test: Boolean
+          })
+
+          defineEmits(['update:test'])
+
+          console.log('test1')
+          console.log('test2')
+          console.log('test3')
+        </script>
+      `,
```

Comment:
@ota-meshi here is the next iteration :)
This logic introduces new lines(but only if these new lines were before)

Reply:
For me, it feels more natural variant. Because it keeps a new line between import and next token as it was before the rearrangement. 
But it may introduce newline between defineProps and defineEmits if they moved in one time

Reply:
Maybe we should try your variant... Don't know) Both have tradeoffs